src/components/TheFloatButtons.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <div class="float" :style="{ bottom: nudge + 'px' }">
+    <transition name="fade">
+      <AppButton
+        v-if="showJump"
+        class="button"
+        icon="angle-up"
+        v-tooltip="'Jump to top of page'"
+        @click="jump()"
+      />
+    </transition>
+    <transition name="fade">
+      <AppButton
+        v-if="showFeedback"
+        class="button"
+        icon="comment"
+        @click="showModal = true"
+        v-tooltip="'Give us feedback on this page!'"
+      />
+    </transition>
+    <AppModal v-model="showModal">
+      <AppFeedbackForm :modal="true" />
+    </AppModal>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import AppModal from "@/components/AppModal.vue";
+import AppFeedbackForm from "@/components/AppFeedbackForm.vue";
+
+// floating button that lets user conveniently jump to top of page
+export default defineComponent({
+  components: {
+    AppModal,
+    AppFeedbackForm,
+  },
+  data() {
+    return {
+      // whether to show jump button
+      showJump: false,
+      // whether to show feedback form button
+      showFeedback: true,
+      // whether to show feedback modal
+      showModal: false,
+      // how much to push buttons upward to make room for footer if in view
+      nudge: 0,
+    };
+  },
+  methods: {
+    update() {
+      // get dimensions of footer
+      const footerEl = document.querySelector("footer");
+      if (!footerEl) return;
+      const footer = footerEl.getBoundingClientRect();
+
+      // if user has scrolled far down enough
+      const downEnough = window.scrollY > window.innerHeight * 0.25;
+
+      // if user has scrolled to bottom of page (and not at top)
+      const atBottom =
+        window.scrollY > 0 &&
+        footer.bottom < footer.height / 2 + window.innerHeight;
+
+      // if user already on dedicated feedback page
+      const onFeedback =
+        ((this.$route.name || "") as string).toLowerCase() === "feedback";
+
+      // show/hide buttons based on scroll
+      this.showJump = downEnough && !atBottom;
+      this.showFeedback = !onFeedback && !atBottom;
+
+      // calculate nudge
+      this.nudge = Math.max(
+        0,
+        footer.height + window.innerHeight - footer.bottom
+      );
+    },
+    jump() {
+      // jump to top of page
+      window.scrollTo(0, 0);
+    },
+  },
+  mounted() {
+    window.addEventListener("scroll", this.update);
+    window.addEventListener("resize", this.update);
+  },
+  updated() {
+    this.update();
```

Comment:
This could be perfectly fine, but I wanted to flag it for the future as a possible performance concern. It might be better to add a watcher on `this.$route` (the only observable value I see accessed in `update()`) that also calls `update()` if the route changes, rather than firing on any update.

Reply:
This was a mistake. Resolved in https://github.com/monarch-initiative/monarch-ui-new/pull/58/commits/3a938df06a8d6b310c30a8f64f1cf690e9e7ad00